### ISSUE TITLE
Bogus signatures in LightClientAttackEvidence

### ISSUE THREAD

**Issue by insumity** [cometbft/cometbft#1749]
### Bug Report (low severity)
At a high level, a malicious validator with more than 1/3 of the voting power could generate `LightClientAttackEvidence` that contains _correct_ signatures but also contains _bogus_ (e.g., unverifiable) signatures. Such `LightClientAttackEvidence` can be added to the _evidence pool_ without an issue if the correct signatures stem from validators that have more than 1/3 of the voting power. As a result when we go through the signatures to slash validators we could go through bogus signatures as well and potentially slash innocent validators.
 
Specifically, the issue has to do in the way that [`LightClientAttackEvidence`](https://github.com/cometbft/cometbft/blob/v0.38.2/proto/tendermint/types/evidence.proto#L28) is verified before being added to the evidence pool.  [`VerifyLightClientAttack`](https://github.com/cometbft/cometbft/blob/v0.38.2/evidence/verify.go#L110) in case of a lunatic attack calls [`VerifyCommitLightTrusting`](https://github.com/cometbft/cometbft/blob/v0.38.2/types/validation.go#L94) with `e.ConflictingBlock.Commit`. Subsequently, `VerifyCommitLightTrusting` calls [`verifyCommitBatch`](https://github.com/cometbft/cometbft/blob/v0.38.2/types/validation.go#L152) or [`verifyCommitSingle`](https://github.com/cometbft/cometbft/blob/v0.38.2/types/validation.go#L265) with the `countAllSignatures` argument  set to `false`. This means that both `verifyCommitBatch` and `verifyCommitSingle` do not go through _all_ the signatures contained in `e.ConflictingBlock.Commit.Signatures` if the signatures that have been verified so far exceed 1/3 (i.e., the [default trust level](https://github.com/cometbft/cometbft/blob/v0.38.2/light/verifier.go#L16)) of the voting power we stop checking when the following condition evaluates to true:
```
if !countAllSignatures && talliedVotingPower > votingPowerNeeded {
```
As a result, a malicious validator (or set of validators) that has more than 1/3 of the voting power could generate `LightClientAttackEvidence` that has all the verifiable signatures at the beginning of the `conflictingBlock.Commit.Signatures` slice and afterwards it has bogus signatures that are not verifiable.
Nevertheless, in such a scenario, the `LightClientAttackEvidence` with the bogus signatures would still be [added](https://github.com/cometbft/cometbft/blob/v0.38.2/evidence/pool.go#L158) in the evidence pool.

When we slash based on this `LightClientAttackEvidence` we [extract the byzantine validators](https://github.com/cometbft/cometbft/blob/v0.38.2/types/evidence.go#L257) that would go through all the signatures:
```
for _, commitSig := range l.ConflictingBlock.Commit.Signatures {
    ...
    _, val := commonVals.GetByAddress(commitSig.ValidatorAddress)
```
And hence we could slash validators that have their `ValidatorAddress` in some of the signatures but those validators have not performed a light client attack. This way, a malicious validator would get slashed, but can also succeed in slashing innocent validators.

**CometBFT version:** Note that the above provided links stem from CometBFT v0.38.2 but I would assume the issue exists since the introduction of `LightClientAttackEvidence`.

Thanks to @jmalicevic and @sergio-mena for looking into this.

**Comment by sergio-mena** [comment]
Thanks for submitting this @insumity
Great find!

**Comment by sergio-mena** [comment]
Closed by #1750, #1806, #1807, #1808, and #1815
